app/Nova/ActionGuards/Keycloak/BlockKeycloakClientGuard.php

```diff
@@ -21,6 +22,10 @@ public function canDo(object $client): bool
             return false;
         }
 
-        return ! $this->cachedKeycloakClientStatus->isClientBlocked($client);
+        try {
+            return ! $this->cachedKeycloakClientStatus->isClientBlocked($client);
+        } catch (KeyCloakApiFailed) {
+            return false;
```

Comment:
Is it not weird to convert an exception here to a boolean? This means it wouldn't be logged in Sentry and it feels like "everything" is normal. I think it's even fine we had the errors inside the admin panel, it was an indication that something was wrong. Preferably those issues are logged with an error and will arrive in Sentry.

Reply:
I wonder if the best approach is to make the combination of integration id and environment unique in the table and still keep the uncaught exception? With that restriction, the initial migration would have failed.